src/com/mambu/apisdk/services/CommentsService.java

```diff
@@ -0,0 +1,100 @@
+package com.mambu.apisdk.services;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import com.google.inject.Inject;
+import com.mambu.apisdk.MambuAPIService;
+import com.mambu.apisdk.exception.MambuApiException;
+import com.mambu.apisdk.util.JSONComment;
+import com.mambu.apisdk.util.MambuEntity;
+import com.mambu.apisdk.util.RequestExecutor.Method;
+import com.mambu.core.shared.model.Comment;
+
+/**
+ * Service class which handles API operations for retrieving and posting comments for Mambu Entities. Posting and
+ * Getting Comments for the following entities are currently supported: Client, Group. LoanAccount, SavingsAccount,
+ * LoanProduct, SavingsProduct, Branch, Centre, User
+ * 
+ * See more details in MBU-8608 - As a Developer, I'd like to GET comments via APIs and MBU-8609 - As a Developer, I
+ * need to POST comments via APIs
+ * 
+ * @author mdanilkis
+ * 
+ */
+
+public class CommentsService extends OwnedEntityService {
```

Comment:
Let's see how this inheritance works further down the road. The problem is that it's hard to compose multiple "services" via inheritance, since you can only inherit from one class. Composition would be preferred here in general and can be easily achieved as well by having an interface and inside the implemented methods just delegate to an implementation from another class.